### ISSUE TITLE
How to render script tags

### ISSUE THREAD

**Issue by mk-pmb** [pugjs/pug#2566]
I found the script block example in the [docs chapter "Plain Text"](http://web.archive.org/web/20161004143245/https://pugjs.org/language/plain-text.html), but I can't use that for [my script](https://gist.github.com/mk-pmb/f884b0a11f72ef5b92212dd7bc54ecc1) because as stated in that chapter, "Plain text can include […]html". Is there an easy way to have pug escape it for me?

**Comment by ForbesLindesay** [comment]
You could try using the cdata filter/jstransformer.

The easiest thing would just be to make it an external script rather than inline.

**Comment by mk-pmb** [comment]
So I installed `jstransformer-cdata` and `jstransformer-cdata-js` and updated my pug file (and gist) to add a script tag with each, re-rendered it, it still doesn't work. I expected that, because [in HTML, CDATA is just a random non-standard tag.](https://mk-pmb.github.io/tryjs/syntax/html/cdata-in.html)

External script files are a valid work-around of course, but you need to make them available via a seperate URI (e.g. save as file in static dir) and they'll add latency to a website.

**Comment by mk-pmb** [comment]
I also tried `:escape-html`, won't work either. I tried making up my own escape function, but none of my ideas worked.

**Comment by ForbesLindesay** [comment]
I don't think this should be solved via a filter.  There is no good universal way to take a snippet of JavaScript and put it inside some HTML.  You can escape `<` and `>` inside strings by replacing it with `\u003C` and `\u003E` respectively.  This will break outside of strings though.  You should not be allowing arbitrary user supplied JavaScript, so shouldn't need to worry about JS like `0</script>/`  If you need to embed user supplied **strings** in JavaScript, you should use js-stringify (we should make sure this is documented somewhere):

```js
var stringify = require('js-stringify');

pug.renderFile(filename, {userSuppliedValue, stringify});
```

```pug
script.
  var foo = #{stringify(userSuppliedValue)};
```

Note that `foo` could still be a string, number, object, array etc. but if it contained strings like `</script>` these will have been properly escaped.
